### PR TITLE
WeatherAPI

### PR DESCRIPTION
**Pull request by AbdullohAkbarov**

### CODE REVIEW

File: Weather/Program.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public static async Task Main(string[] args)
             else
             {
                 Console.WriteLine(response.Error);
-            }
```
Comment (denja2) [inline Weather/Program.cs:32]: Why you changed that?

File: WeatherAPI/Controllers/WeatherController.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace WeatherAPI.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class WeatherController : Controller
+    {
+        private WeatherService _service;
+        private readonly IConfiguration _configuration;
+        private ILog log;
```
Comment (denja2) [inline WeatherAPI/Controllers/WeatherController.cs]: Not neccessary

File: WeatherAPI/Controllers/WeatherController.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        public WeatherController(IConfiguration configuration)
+        {
+            _configuration = configuration;
+            _service = new WeatherService(new WeatherProvider(new HttpClient(), _configuration.GetValue<string>("url"), _configuration.GetValue<string>("appid")), new CityValidator());
```
Comment (denja2) [inline WeatherAPI/Controllers/WeatherController.cs]: Use DI.

**Review by denja2 — CHANGES_REQUESTED** [review]

File: Weather/Program.cs (hunk 1 above)
Comment (AbdullohAkbarov) [reply Weather/Program.cs:32]: deleted the whitespaces

File: WeatherAPI/Controllers/WeatherController.cs (hunk 3 above)
Comment (AbdullohAkbarov) [reply WeatherAPI/Controllers/WeatherController.cs]: used

File: WeatherAPI/Controllers/WeatherController.cs (hunk 2 above)
Comment (AbdullohAkbarov) [reply WeatherAPI/Controllers/WeatherController.cs]: fixed

File: WeatherAPI/Controllers/WeatherController.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        [Route("GetWeather")]
+        [HttpGet]
+        public async Task<string> GetWeather(string city)
+        {
+            var response = await _service.GetCurrentWeatherAsync(city);
+            if (response.IsSuccess is false)
+            {
+                //log.Error(response.Error);
```
Comment (denja2) [inline WeatherAPI/Controllers/WeatherController.cs]: Remove such comments

File: WeatherAPI/Startup.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+                app.UseSwagger();
+                app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "WeatherAPI v1"));
+            }
+
+            app.UseRouting();
+
+            app.UseAuthorization();
```
Comment (denja2) [inline WeatherAPI/Startup.cs]: What is the reason for use it here?

File: WeatherDAL/WeatherProvider.cs [hunk 6]
```diff
@@ -11,15 +11,15 @@ namespace WeatherDAL
 {
     public class WeatherProvider : IWeatherProvider
     {
-        private readonly HttpClient client;
+        private readonly HttpClient _client;
         private readonly string appid;
```
Comment (denja2) [inline WeatherDAL/WeatherProvider.cs]: _appId

File: WeatherDAL/WeatherProvider.cs [hunk 7]
```diff
@@ -11,15 +11,15 @@ namespace WeatherDAL
...
         private readonly string url;
```
Comment (denja2) [inline WeatherDAL/WeatherProvider.cs]: _url

**Review by denja2 — CHANGES_REQUESTED** [review]

File: WeatherDAL/WeatherProvider.cs (hunk 6 above)
Comment (AbdullohAkbarov) [reply WeatherDAL/WeatherProvider.cs]: fixed

File: WeatherDAL/WeatherProvider.cs (hunk 7 above)
Comment (AbdullohAkbarov) [reply WeatherDAL/WeatherProvider.cs]: fixed

File: WeatherAPI/Controllers/WeatherController.cs (hunk 4 above)
Comment (AbdullohAkbarov) [reply WeatherAPI/Controllers/WeatherController.cs]: fixed

File: WeatherAPI/Startup.cs (hunk 5 above)
Comment (AbdullohAkbarov) [reply WeatherAPI/Startup.cs]: app.userouting() is used to route requests on middleware
app.useauthorization() is used to authorizes a user to access secure resources

Comment (denja2) [reply WeatherAPI/Startup.cs]: Do we have secured resources here?

Comment (AbdullohAkbarov) [reply WeatherAPI/Startup.cs]: no cause I didn't add any kind of authentication or authorization

**Pull request merged**